src/features/ExtensionCommands.ts

```diff
@@ -388,44 +390,106 @@ export class ExtensionCommandsFeature implements IFeature {
-        if (!this.findTextDocument(this.normalizeFilePath(saveFileDetails.filePath))) {
-            return EditorOperationResponse.Completed;
+        // Try to interpret the filepath as a URI, defaulting to "file://" if we don't succeed
+        let currentFileUri: vscode.Uri;
+        if (saveFileDetails.filePath.startsWith("untitled") || saveFileDetails.filePath.startsWith("file")) {
+            currentFileUri = vscode.Uri.parse(saveFileDetails.filePath);
+        } else {
+            currentFileUri = vscode.Uri.file(saveFileDetails.filePath);
         }
 
-        // If no newFile is given, just save the current file
-        if (!saveFileDetails.newPath) {
-            const doc = await vscode.workspace.openTextDocument(saveFileDetails.filePath);
-            if (doc.isDirty) {
-                await doc.save();
-            }
-
-            return EditorOperationResponse.Completed;
+        let newFileAbsolutePath: string;
+        switch (currentFileUri.scheme) {
+            case "file":
+                // If the file to save can't be found, just complete the request
+                if (!this.findTextDocument(this.normalizeFilePath(currentFileUri.fsPath))) {
+                    return EditorOperationResponse.Completed;
+                }
+
+                // If no newFile is given, just save the current file
+                if (!saveFileDetails.newPath) {
+                    const doc = await vscode.workspace.openTextDocument(currentFileUri.fsPath);
+                    if (doc.isDirty) {
+                        await doc.save();
+                    }
+
+                    return EditorOperationResponse.Completed;
+                }
+
+                // Make sure we have an absolute path
+                if (path.isAbsolute(saveFileDetails.newPath)) {
+                    newFileAbsolutePath = saveFileDetails.newPath;
+                } else {
+                    // If not, interpret the path as relative to the current file
+                    newFileAbsolutePath = path.join(path.dirname(currentFileUri.fsPath), saveFileDetails.newPath);
+                }
+
+                await this.saveDocumentContentToAbsolutePath(currentFileUri, newFileAbsolutePath);
+                return EditorOperationResponse.Completed;
+
+            case "untitled":
+                // We need a new name to save an untitled file
+                if (!saveFileDetails.newPath) {
+                    return EditorOperationResponse.Completed;
+                }
+
+                // Make sure we have an absolute path
+                if (path.isAbsolute(saveFileDetails.newPath)) {
+                    newFileAbsolutePath = saveFileDetails.newPath;
+                } else {
+                    // If not, interpret the path as relative to the workspace root
+                    const workspaceRootUri = vscode.workspace.workspaceFolders[0].uri;
+                    // We don't saving to a non-file URI-schemed workspace
```

Comment:
nit pick: comment language:
"we don't saving" should be "we don't support saving"